src/GADM.jl

```diff
@@ -52,86 +53,85 @@ function dataread(path)
     !isnothing(data) ? data : throw(ArgumentError("failed to read data from disk"))
 end
 
+"""
+    getdataset(country)
+
+Downloads and extracts dataset of the given country code
+"""
+function getdataset(country)
+    isvalidcode(country) || throw(ArgumentError("please provide standard ISO 3 country codes"))
+    data = country |> download |> dataread
+end
+
 """
     getlayer(data, level) 
 
 Get layer of the desired `level` from the `data`.
 """
 function getlayer(data, level)
     nlayers = ArchGDAL.nlayer(data)
-    for l=0:nlayers-1
+    for l = 0:nlayers - 1
         layer = ArchGDAL.getlayer(data, l)
         lname = ArchGDAL.getname(layer)
         llevel = last(split(lname, "_"))
         string(level) == llevel && return layer
     end
-    throw(ArgumentError("valid levels are 0...$(nlayers-1)"))
+    throw(ArgumentError("asked for level $(level), valid levels are 0-$(nlayers - 1)"))
 end
 
 """
-    get(country, subregions...)
+    get(country, subregions...;children=false)
+Returns a Tables.jl columntable for the requested region
+Geometry of the region(s) can be accessed with the key `geom`
+The geometries are GeoInterface compliant Polygons/MultiPolygons.
 
-Returns the MULTIPOLYGON data for the requested region.
-
-Input: ISO 3 country code, and further subdivisions.
+1. country: ISO 3166 Alpha 3 country code  
+2. subregions: Full official names in hierarchial order (provinces, districts, etc.)  
+3. children: When true, function returns two columntables -> parent, children.  
+Eg. when children is set true when querying just the country,
+second return parameter are the states/provinces.  
 
 ## Examples  
   
 ```julia
-get("IND")
-get("IND", "Uttar Pradesh")
-get("IND", "Uttar Pradesh", "Lucknow")
+data = get("IND") # columntable of size 1, data of India's borders
+parent, children = get("IND", "Uttar Pradesh";children=true)
+# parent -> state data, children -> table of all districts inside Uttar Pradesh
```

Comment:
I think this comment should come before the actual code line? The same for the previous example.